compiler/rustc_const_eval/src/interpret/validity.rs

```diff
@@ -496,22 +553,28 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValidityVisitor<'rt, 'mir, '
         let ty = value.layout.ty;
         match ty.kind() {
             ty::Bool => {
-                let value = self.read_scalar(value, "a boolean")?;
+                let value = self.read_scalar(value, "bool")?;
                 try_validation!(
                     value.to_bool(),
                     self.path,
-                    InvalidBool(..) =>
-                        { "{:x}", value } expected { "a boolean" },
+                    InvalidBool(..) => expected("bool"),
+                    {
+                        found = "custom",
+                        value = format!("{value:x}"),
+                    }
```

Comment:
I think we need to work on the syntax here.  The order of "$value \expected a boolean" was deliberate since that is how it is printed. Also now it looks like the case ends after the `,` and then the curly brace part is completely unrelated, that's very confusing.

What do all these strings and names even mean?

Reply:
They are located in `messages.ftl` but I agree that this should be improved. I will look into making this typed instead of using closures and macros.

Reply:
Also once you figured out the interface to the fluent messages we need to work on the macro syntax. So let's keep this subthread open for that purpose.

Reply:
I've replaced the previous macro stuff with enum variants which should make it very straight forward.